tracer/src/Datadog.Trace/AppSec/Waf/Waf.cs

```diff
@@ -51,20 +50,69 @@ internal Waf(IntPtr wafHandle, WafLibraryInvoker wafLibraryInvoker)
         /// <param name="rulesFile">can be null, means use rules embedded in the manifest </param>
         /// <param name="rulesJson">can be null. RemoteConfig rules json. Takes precedence over rulesFile </param>
         /// <returns>the waf wrapper around waf native</returns>
-        internal static InitializationResult Create(WafLibraryInvoker wafLibraryInvoker, string obfuscationParameterKeyRegex, string obfuscationParameterValueRegex, string? rulesFile = null, string? rulesJson = null)
+        internal static InitOrUpdateResult Create(WafLibraryInvoker wafLibraryInvoker, string obfuscationParameterKeyRegex, string obfuscationParameterValueRegex, string? rulesFile = null, string? rulesJson = null)
         {
             var wafConfigurator = new WafConfigurator(wafLibraryInvoker);
-            InitializationResult initializationResult;
+            InitOrUpdateResult initOrUpdateResult;
             if (!string.IsNullOrEmpty(rulesJson))
             {
-                initializationResult = wafConfigurator.ConfigureFromRemoteConfig(rulesJson, obfuscationParameterKeyRegex, obfuscationParameterValueRegex);
+                initOrUpdateResult = wafConfigurator.ConfigureFromRemoteConfig(rulesJson, obfuscationParameterKeyRegex, obfuscationParameterValueRegex);
             }
             else
             {
-                initializationResult = wafConfigurator.Configure(rulesFile, obfuscationParameterKeyRegex, obfuscationParameterValueRegex);
+                initOrUpdateResult = wafConfigurator.Configure(rulesFile, obfuscationParameterKeyRegex, obfuscationParameterValueRegex);
             }
 
-            return initializationResult;
+            return initOrUpdateResult;
+        }
+
+        public InitOrUpdateResult UpdateRules(string rules)
+        {
+            var rulesetInfo = new DdwafRuleSetInfo();
+            try
+            {
+                var argCache = new List<Obj>();
```

Comment:
If we're not using this to dispose intermediate objects, probably better to pass null, see earlier comments.

Reply:
*very* good catch thank you!!! missed disposing args that we indeed need to dispose all the time, as it's always  the unmanged`Update` called on the waf from now, so fixed [here](https://github.com/DataDog/dd-trace-dotnet/blob/89c9a966f7cebca21adf83a8c4eace656995f556/tracer/src/Datadog.Trace/AppSec/Waf/Waf.cs#L103)

for any type of update, we update the handle and dispose all related items